### PR TITLE
Improves authentication error handling

### PR DESCRIPTION
**Pull request by jayrmotta**
**Problem**: Messages coming from the backend were being overwritten by our frontend and therefore delivering misleading messages

**Solution**: Bypass messages coming from the backend for a given pattern but return a friendly message for unexpected errors like when our backend is down or failing

### CODE REVIEW

File: src/login/ErrorMessages.spec.js [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
     );
   });
 
+  it('should ensure authentication messages coming from the backend are shown', () => {
+    expect(
+      lookupErrorMessage('invalid_grant', 'Backend error message'),
+    ).toEqual('Backend error message');
+  });
+
   it('should ensure default message is used when no friendly message is found', () => {
```
Comment (ttalviste) [inline src/login/ErrorMessages.spec.js]: Looking into the previous test. There was provided a fallback messages when the lookup failed. At the moment you override the fallback message with a new message and in my mind key information will be lost that could be sent to the customer support.

Comment (jayrmotta) [reply src/login/ErrorMessages.spec.js]: But on the saga that uses this error message lookup it does trigger a sentry tracking with the errors that happened, so for the purpose of tracking I think we are fine.

File: src/login/ErrorMessages.js [hunk 2]
```diff
@@ -1,13 +1,20 @@
 export default function(errorCode, defaultMessage) {
```
[the remark this answers is not in the document]
Comment (jayrmotta) [reply src/login/ErrorMessages.js]: You're right, I'll send a commit improving this.

File: src/login/ErrorMessages.spec.js (hunk 1 above)
Comment (ttalviste) [reply src/login/ErrorMessages.spec.js]: True. Nevertheless, how do you see that an error will be resolved when there is no info shown to the customer, e.g on the screen or console?

Comment (jayrmotta) [reply src/login/ErrorMessages.spec.js]: The information is exhibited on the screen, the only errors we are suppressing here are the ones that we don't think the user should see at all like internal server errors or something like that.

**Pull request merged**